File: software/examples/java/ExampleInterrupt.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletIO16;

public class ExampleInterrupt {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIO16 io = new BrickletIO16(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add interrupt listener
		io.addInterruptListener(new BrickletIO16.InterruptListener() {
			public void interrupt(char port, short interruptMask, short valueMask) {
				System.out.println("Port: " + port);
				System.out.println("Interrupt Mask: " + Integer.toBinaryString(interruptMask));
				System.out.println("Value Mask: " + Integer.toBinaryString(valueMask));
				System.out.println("");
			}
		});

		// Enable interrupt on pin 2 of port A
		io.setPortInterrupt('a', (short)(1 << 2));

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_interrupt.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/io16_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your IO-16 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	io, _ := io16_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	io.RegisterInterruptCallback(func(port rune, interruptMask uint8, valueMask uint8) {
		fmt.Printf("Port: %c\n", port)
		fmt.Printf("Interrupt Mask: %b\n", interruptMask)
		fmt.Printf("Value Mask: %b\n", valueMask)
		fmt.Println()
	})

	// Enable interrupt on pin 2 of port A
	io.SetPortInterrupt('a', 1<<2)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
